core/crypto/randomness/impl/randomness_provider_imp.cpp

```diff
@@ -12,23 +12,24 @@ namespace fc::crypto::randomness {
 
   Randomness RandomnessProviderImpl::deriveRandomness(DomainSeparationTag tag,
                                                       Serialization s) {
-    return deriveRandomness(tag, std::move(s), ChainEpoch(-1));
+    return deriveRandomness(tag, std::move(s), -1);
   }
 
   Randomness RandomnessProviderImpl::deriveRandomness(DomainSeparationTag tag,
                                                       Serialization s,
                                                       const ChainEpoch &index) {
-    return deriveRandomnessInternal(tag, std::move(s), index);
+    return deriveRandomnessInternal(
+        tag, std::move(s), index.convert_to<int64_t>());
   }
 
   Randomness RandomnessProviderImpl::deriveRandomnessInternal(
-      DomainSeparationTag tag, Serialization s, const ChainEpoch &index) {
+      DomainSeparationTag tag, Serialization s, int64_t index) {
```

Comment:
Why not leave ChainEpoch as it was before?
Ofc, when ChainEpoch is uint64_t, it would be better to remove const &.

Reply:
Because this function is called with -1 as ChainEpoch.